File: Arrays/sort_colors.cpp
class Solution {
public:
	vector<int> sortColors(vector<int>& nums) {
		// Your code goes here
        int low = 0;
        int mid = 0;
        int high = nums.size()-1;
        while(mid<=high){
            if(nums[mid]==2){
                int t = nums[high];
                nums[high] = nums[mid];
                nums[mid] = t;
                high--;
            }else if(nums[mid]==0){
                int t = nums[low];
                nums[low] = nums[mid];
                nums[mid] = t;
                low++;
                mid++;
            }else{
                mid++;
            }
        }
        return  nums;
	}
};

File: Arrays/sort_colors.js
class Solution {
	/** 
	 * @param {number[]} nums
	 * @return {number[]}
	 */
	sortColors(nums) {
		// Your code goes 
        let low = 0;
        let mid = 0;
        let high = nums.length-1;
        while(mid<=high){
            if(nums[mid]==0){
                let t= nums[low];
                nums[low] = nums[mid];
                nums[mid] = t;
                low++;
                mid++;
            }else if(nums[mid]==2){
                let t = nums[high];
                nums[high] = nums[mid];
                nums[mid] = t;
                high--;
            }else{
                mid++;
            }
        }
        return nums;
	}
};
